configure.py

```diff
@@ -1662,6 +1662,9 @@ def dynamic_linker_option():
 curdir = os.getcwd()
 user_cflags = args.user_cflags + f" -ffile-prefix-map={curdir}=."
 
+# Since gcc 13, libgcc doesn't need the exception workaround
+user_cflags += ' -DNO_EXCEPTION_INTERCEPT -DSEASTAR_NO_EXCEPTION_HACK'
```

Comment:
probably unrelated, but can we fix this on Seastar's end? because Seastar only supports the latest two major versions of compiler. when it comes to GCC, the latest two major versions are GCC-13 and GCC-14. both of them include the fix in libgcc of https://github.com/gcc-mirror/gcc/commit/6e80a1d164d1f996ad08a512c000025a7c2ca893#diff-03e90e365ad13835bc93337b2a1ccc264c121395d6d1a293c13948c8faf73d93R169 .

in general, user installs the runtime along with the compiler, presumably, we can assume that GCC-13 comes with libgcc-13. so i guess, we should can just either just drop the workaround, or define there two macros by default? if we really want to check the used runtime, probably we can use the proposed approach at https://github.com/scylladb/seastar/issues/2479#issuecomment-2428150691 ?

Reply:
AFAICS with NO_EXCEPTION_INTERCEPT we'll also miss the `memory_cpp_exceptions` metrics accounting. Is that what we want?

Reply:
ahh, i forgot that we also had non-multi-config-generator use case, so we need 
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 978b21686f..7b93b61a61 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,6 +95,10 @@ else()
     set(Seastar_SCHEDULING_GROUPS_COUNT 16 CACHE STRING "" FORCE)
     set(Seastar_UNUSED_RESULT_ERROR ON CACHE BOOL "" FORCE)
     add_subdirectory(seastar)
+    target_compile_definitions (seastar
+      PRIVATE
+        NO_EXCEPTION_INTERCEPT
+        SEASTAR_NO_EXCEPTION_HACK)
 endif()
 
 set(ABSL_PROPAGATE_CXX_STD ON CACHE BOOL "" FORCE)
```

Reply:
> AFAICS with NO_EXCEPTION_INTERCEPT we'll also miss the `memory_cpp_exceptions` metrics accounting. Is that what we want?

No.

Reply:
created #21233